File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys

inputs = sys.stdin.readline().rstrip()
bomb = sys.stdin.readline().rstrip()

stack = []

for i in range(len(inputs)):
    stack.append(inputs[i])
    if ''.join(stack[-len(bomb):]) == bomb:
        for i in range(len(bomb)):
            stack.pop()

if stack:
    print(''.join(stack))
else:
    print('FRULA')

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "./example.txt";

let input = fs.readFileSync(filePath).toString().split("\n");
let [string, bomb] = input;
let bombLength = bomb.length;
let stack = [];
let reverseString = [...string].reverse();

while (reverseString.length !== 0) {
  stack.push(reverseString.pop());
  if (
    stack.length >= bombLength &&
    stack.slice(stack.length - bombLength, stack.length).join("") === bomb
  ) {
    stack.splice(stack.length - bombLength, bombLength);
  }
}

if (stack.length === 0) {
  console.log("FRULA");
} else {
  console.log(stack.join(""));
}
